LogMonitor/src/LogMonitor/EtwMonitor.cpp

```diff
@@ -705,6 +705,69 @@ EtwMonitor::OnRecordEvent(
     return status;
 }
 
+///
+/// Format ETW eventlog into a JSON output
+///
+std::wstring etwJsonFormat(EtwLogEntry* pLogEntry)

```

Comment:
Why are we passing a pointer here?

Reply:
For memory efficiency and correctness, passing by value create a copy of the struct (hence extra space). also, if you modify that copy, the original one is still unmodified.